### PR TITLE
Feature/listview multiple templates

### PR DESCRIPTION
**Pull request by arthurbleilzup**
**- What I did**
Implemented the `doTemplateRender` on the Dynamic Lists

Closes: #338 and [ZupIT/beagle#1523](https://github.com/ZupIT/beagle/issues/1523)

### CODE REVIEW

File: src/components/dynamic-lists/dynamic-list.component.ts [hunk 1]
```diff
@@ -128,29 +124,59 @@ export class DynamicListComponent
...
+    if (!Array.isArray(this.dataSource)) return
+
+    const viewContentManager = this.getViewContentManager()
+
+    const element = viewContentManager.getElement()
+    if (!element) return logger.error('The beagle:listView element was not found.')
+
+    const templatesRaw: BeagleListViewInterface['templates'] = 
+        viewContentManager.getElement().templates
+    if (!this.template &&
```
Comment (Tiagoperes) [inline src/components/dynamic-lists/dynamic-list.component.ts:137]:
```suggestion
    const hasTemplate = this.template ||  (templatesRaw && Array.isArray(templatesRaw) && templatesRaw.length)
    if (!hasTemplate) {
```

File: src/components/dynamic-lists/dynamic-list.component.ts [hunk 2]
```diff
@@ -128,29 +124,59 @@ export class DynamicListComponent
...
+
+    const viewContentManager = this.getViewContentManager()
+
+    const element = viewContentManager.getElement()
+    if (!element) return logger.error('The beagle:listView element was not found.')
+
+    const templatesRaw: BeagleListViewInterface['templates'] = 
+        viewContentManager.getElement().templates
+    if (!this.template && 
+        (!templatesRaw || !Array.isArray(templatesRaw) || templatesRaw.length === 0)) {
```
Comment (Tiagoperes) [inline src/components/dynamic-lists/dynamic-list.component.ts:138]:
```suggestion

```

File: src/components/dynamic-lists/dynamic-list.component.ts [hunk 3]
```diff
@@ -128,29 +124,59 @@ export class DynamicListComponent
...
+      return logger.error('The beagle:listView requires a template or multiple templates to be rendered!')
+    }
+
+    const componentTag = element._beagleComponent_.toLowerCase()
+    const deprecatedTemplate = { case: false, view: this.template }
```
Comment (Tiagoperes) [inline src/components/dynamic-lists/dynamic-list.component.ts:143]: Considering all lines below, do the same as you did for React.

**Review by Tiagoperes — CHANGES_REQUESTED** [review]

File: src/components/dynamic-lists/dynamic-list.component.ts [hunk 4]
```diff
@@ -128,29 +124,59 @@ export class DynamicListComponent
...
 
+    if (!Array.isArray(this.dataSource)) return
+
+    const viewContentManager = this.getViewContentManager()
+
+    const element = viewContentManager.getElement()
+    if (!element) return logger.error('The beagle:listView element was not found.')
+
+    const templatesRaw: BeagleListViewInterface['templates'] = 
+        viewContentManager.getElement().templates
```
[the remark this answers is not in the document]
Comment (arthurbleilzup) [reply src/components/dynamic-lists/dynamic-list.component.ts]: Implemented as suggested

File: src/components/dynamic-lists/dynamic-list.component.ts (hunk 1 above)
Comment (arthurbleilzup) [reply src/components/dynamic-lists/dynamic-list.component.ts]: Implemented as suggested

File: src/components/dynamic-lists/dynamic-list.component.ts (hunk 2 above)
Comment (arthurbleilzup) [reply src/components/dynamic-lists/dynamic-list.component.ts]: Implemented as suggested

File: src/components/dynamic-lists/dynamic-list.component.ts (hunk 3 above)
Comment (arthurbleilzup) [reply src/components/dynamic-lists/dynamic-list.component.ts]: Implemented as suggested

**Review by Tiagoperes — APPROVED** [review]

**Review by Tiagoperes — APPROVED** [review]

**Pull request merged**